app/controllers/admin/users_controller.rb

```diff
@@ -91,20 +101,47 @@ def create
             m.course = @course
           }
           raise ActiveRecord::Rollback unless membership.save
+          
+          # if user is guest, create a entry under guest database as well
+          if params[:user]["course_role"] == '3'
+            guest = GuestUsersDetail.new { |g|
+              g.user_id = @the_user.id
+              g.course_id = @course.id
+              g.hash_string = params[:user]["id_string"]
+              g.assignment_id = 0
+            }
+            raise ActiveRecord::Rollback unless guest.save
+          end
         end
       end
     end
 
-    # Check for errors and render view
-    if @the_user.errors.empty? and @the_user.save
-      if @existing_user
-        redirect_to admin_users_url, notice: "User was successfully added 
-        to #{@course.code}."
+    @user = User.find_by_id(session[:user_id]) 
+
+    if (@user.is_admin)
+      # Check for errors and render view
+      if @the_user.errors.empty? and @the_user.save
+        if @existing_user
+          redirect_to admin_users_url, notice: "User was successfully added 
+          to #{@course.code}."
+        else 
+          redirect_to admin_users_url, notice: 'User was successfully created.'
+        end
       else
-        redirect_to admin_users_url, notice: 'User was successfully created.'
+        render action: "new"
+      end
+    else 
```

Comment:
L119-143 seem repetitive.  can collapse into one block?

Reply:
My bad as well. The lines are redundant and I have removed them.